conda/common/logic.py

```diff
@@ -540,8 +789,8 @@ def sum_val(sol, odict):
                     if done:
                         break
                 self.m = m_orig
-                if len(self.clauses) > nz:
-                    self.clauses = self.clauses[:nz]
+                if self.save_state() != saved_state:
```

Comment:
We know we only add clauses but never remove any, apart from when we reset them here. So testing on equality is sufficient here. This way we don't necessarily have to use the clause count but can also use some general "state" we compare to. Helpful for the `ClauseArray` implementation.